src/interceptors/ClientRequest/agents.ts

```diff
@@ -19,33 +19,28 @@ interface MockAgentOptions {
   onResponse: MockHttpSocketResponseCallback
 }
 
-export class MockAgent extends http.Agent {
-  private customAgent?: http.RequestOptions['agent']
-  private onRequest: MockHttpSocketRequestCallback
-  private onResponse: MockHttpSocketResponseCallback
+export function createHttpAgent(options: MockAgentOptions): http.Agent {
```

Comment:
Mandatory change. Otherwise, we aren't respecting the `options.customAgent` enough (ignoring its `keepalive`, max sockets, etc.).